api/src/main/java/javax/enterprise/inject/spi/builder/InjectionPointConfigurator.java

```diff
@@ -0,0 +1,187 @@
+import java.util.Set;
+
+/**
+ * This API is an helper to configure a new {@link InjectionPoint} instance.
+ * CDI container must provides an implementation of this interface.
+ *
+ * This builder is not thread safe and shall not be used concurrently.
+ *
+ * @see ProcessInjectionPoint#configureInjectionPoint()
+ * @author Antoine Sabot-Durand
+ * @since 2.0
+ */
+public interface InjectionPointConfigurator {
+
+
+    /**
+     * Read the InjectionPoint information from the given {@link Field}.
+     * All relevant information is overwritten.
+     *
+     * @param field defining the InjectionPoint.
+     * @return self
+     */
+    InjectionPointConfigurator read(Field field);
+
+    /**
+     * Read the InjectionPoint information from the given {@link Method}.
+     * All relevant information is overwritten.
+     *
+     * @param method defining the InjectionPoint.
+     * @param param  the parameter's index in the Method defining the InjectionPoint
+     * @return self
+     */
+    InjectionPointConfigurator read(Method method, int param);
+
+    /**
+     * Read the InjectionPoint information from the given {@link Constructor}.
+     * All relevant information is overwritten.
+     *
+     * @param constructor defining the InjectionPoint.
+     * @param param       the parameter's index in the Constructor defining the InjectionPoint
+     * @return self
+     */
+    InjectionPointConfigurator read(Constructor<?> constructor, int param);
+
+    /**
+     * Read the InjectionPoint information from the given {@link AnnotatedField}.
+     * All relevant information is overwritten.
+     *
+     * @param field defining the InjectionPoint
+     * @return self
+     */
+    InjectionPointConfigurator read(AnnotatedField<?> field);
+
+    /**
+     * Read the InjectionPoint information from the given {@link AnnotatedParameter}.
+     * All relevant information is overwritten.
+     *
+     * @param param defining the InjectionPoint
+     * @return self
+     */
+    InjectionPointConfigurator read(AnnotatedParameter<?> param);
+
+    /**
+     * Read the InjectionPoint information from the given {@link InjectionPoint}.
+     * All relevant information is overwritten.
+     *
+     * @param injectionPoint the InjectionPoint to get information from
+     * @return self
+     */
+    InjectionPointConfigurator read(InjectionPoint injectionPoint);
+
+    /**
+     * Set the required {@link Type} (that will be used during typesafe resolution)
+     * of the InjectionPoint to build.
+     *
+     * @param requiredType for the InjectionPoint to build
+     * @return self
+     */
+    InjectionPointConfigurator type(Type requiredType);
+
+    /**
+     *
+     * Add the qualifier to the InjectionPoint to build
+     * If the builder declares the {@link Default} qualifier, it's automatically removed.
+     *
+     * @param qualifier the qualifier to add
+     * @return self
+     */
+    InjectionPointConfigurator addQualifier(Annotation qualifier);
```

Comment:
This is little bit problematic. When I add some additional qualifier during PIP the processed IP still have also `@Default` which is not desirable. Shouldn't we go only with "replaceable" methods in this case??